lib/internal/process/promises.js

```diff
@@ -1,20 +1,12 @@
 'use strict';
 
-const promiseRejectEvent = process._promiseRejectEvent;
-const hasBeenNotifiedProperty = new WeakMap();
-const promiseToGuidProperty = new WeakMap();
-const pendingUnhandledRejections = [];
-let lastPromiseId = 1;
-
-exports.setup = setupPromises;
-
-function getAsynchronousRejectionWarningObject(uid) {
-  return new Error('Promise rejection was handled ' +
-                   `asynchronously (rejection id: ${uid})`);
-}
-
-function setupPromises(scheduleMicrotasks) {
-  let deprecationWarned = false;
+exports.setup = function setup(scheduleMicrotasks) {
+  const promiseRejectEvent = process._promiseRejectEvent;
+  const hasBeenNotifiedProperty = new Map();
```

Comment:
Can you explain why a regular `Map` is sufficient here?

A WeakMap was used for a very specific reason in order to not interfere with GC.

On a related note, is there any way we can test the unhandled rejection tracking code does not leak memory?

Reply:
(I realize it doesn't do the same thing exactly and tracking happens in C++, but still)

Reply:
`WeakMap`s can't be enumerated, but no enumeration is done here; when `rejectionHandled` is called there is still a strong reference to the promise (the `promise` argument) so it should still exist in the `WeakMap` 🤔

Not sure why a `Map` is needed either; it just creates the need for `.delete` calls.

Reply:
The reason why I changed it is that I can not see a reason to use `WeakMap`. All entries are either going to be handled and in that case explicitly deleted or they result in a unhandled rejection that will now terminate the process. Using the WeakMap has a performance overhead and the GC has to do more work.